src/main/java/com/ververica/flink/table/gateway/rest/result/ResultSet.java

```diff
@@ -85,22 +97,96 @@ public boolean equals(Object o) {
 			return false;
 		}
 		ResultSet resultSet = (ResultSet) o;
-		return columns.equals(resultSet.columns) &&
+		return resultKind.equals(resultSet.resultKind) &&
+			columns.equals(resultSet.columns) &&
 			data.equals(resultSet.data) &&
 			Objects.equals(changeFlags, resultSet.changeFlags);
 	}
 
 	@Override
 	public int hashCode() {
-		return Objects.hash(columns, data, changeFlags);
+		return Objects.hash(resultKind, columns, data, changeFlags);
 	}
 
 	@Override
 	public String toString() {
 		return "ResultSet{" +
-			"columns=" + columns +
+			"resultKind=" + resultKind +
+			", columns=" + columns +
 			", data=" + data +
 			", changeFlags=" + changeFlags +
 			'}';
 	}
+
+	public static Builder builder() {
+		return new Builder();
+	}
+
+	/**
+	 * Builder for {@link ResultSet}.
+	 */
+	public static class Builder {
+		private ResultKind resultKind = null;
+		private List<ColumnInfo> columns = null;
+		private List<Row> data = null;
```

Comment:
Does it make more sense to set the default value to `Collections.emptyList()`?

Reply:
data should be not null in ResultSet, and Builder#data(Row ... data) supports variable arguments, `builder.data()` can easily represent empty data.